pkg/virt-handler/virtwrap/manager.go

```diff
@@ -328,21 +327,14 @@ func VMNamespaceKeyFunc(vm *v1.VM) string {
 	return domName
 }
 
-func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) (*api.DomainSpec, error) {
-	var wantedSpec api.DomainSpec
-	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
-
-	if len(mappingErrs) > 0 {
-		return nil, errors.NewAggregate(mappingErrs)
-	}
-
+func (l *LibvirtDomainManager) SyncVM(vm *v1.VM, domDesign *designer.DomainDesign) (*api.DomainSpec, error) {
```

Comment:
Right the API calls are done outside, missed that. Still the mapping to the domain xml is also done outside of SyncVM. I would expect us to first fully populate the VM object, validate it from the cluster perspective, and then pass it to SyncVM, where it is transformed. So all cluster  related logic on the controller side, all libvirt related logic in the manager. I would also prefer to pass in the secrets as k8s objects, and transform them to libvirt objects in the manager.